### ISSUE TITLE
pyopenms does not built due to Recursion in std::vector<T> in ChromatogramSettings

### ISSUE THREAD

**Issue by mwalzer** [OpenMS/OpenMS#1719]
on current branch develop ( 58fa98b ) with linux & python 2.7.6
how can this be a wrapper failure when it builds on other OS (incl. linux)?

**Comment by mwalzer** [comment]
```
INFO:root:create wrapper for class ChromatogramSettings
INFO:root:   create wrapper for setInstrumentSettings ('void setInstrumentSettings(InstrumentSettings instrument_settings)')
INFO:root:   create wrapper for getDataProcessing ('libcpp_vector[shared_ptr[DataProcessing]] getDataProcessing()')
INFO:root:   create wrapper for setNativeID ('void setNativeID(String native_id)')
INFO:root:   create wrapper for getProduct ('Product getProduct()')
INFO:root:   create wrapper for setSourceFile ('void setSourceFile(SourceFile source_file)')
INFO:root:   create wrapper for setComment ('void setComment(String comment)')
INFO:root:   create wrapper for setPrecursor ('void setPrecursor(Precursor precursor)')
INFO:root:   create wrapper for setProduct ('void setProduct(Product p)')
INFO:root:   create wrapper for setDataProcessing ('void setDataProcessing(libcpp_vector[shared_ptr[DataProcessing]])')
Traceback (most recent call last):
  File "create_cpp_extension.py", line 66, in <module>
    None)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/Main.py", line 190, in create_wrapper_code
    inc_dirs = autowrap.generate_code(decls, instance_map, out, False, manual_code, cimports)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/__init__.py", line 63, in generate_code
    gen.create_pyx_file(debug)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 171, in create_pyx_file
    create_for(ResolvedClass, self.create_wrapper_for_class)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 167, in create_for
    method(resolved)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 300, in create_wrapper_for_class
    codes = self.create_wrapper_for_method(r_class, name, methods)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 412, in create_wrapper_for_method
    code = self.create_wrapper_for_nonoverloaded_method(cdcl, py_name, methods[0])
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 554, in create_wrapper_for_nonoverloaded_method
    method)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/CodeGenerator.py", line 457, in _create_fun_decl_and_input_conversion
    conv_code, call_as, cleanup = converter.input_conversion(t, n, arg_num)
  File "/usr/local/lib/python2.7/dist-packages/autowrap/ConversionProvider.py", line 1124, in input_conversion
    "Recursion in std::vector<T> is not implemented for other STL methods and wrapped template arguments")
Exception: Recursion in std::vector<T> is not implemented for other STL methods and wrapped template arguments
make[3]: *** [src/pyOpenMS/CMakeFiles/pyopenms_create_cpp] Error 1
make[2]: *** [src/pyOpenMS/CMakeFiles/pyopenms_create_cpp.dir/all] Error 2
make[1]: *** [src/pyOpenMS/CMakeFiles/pyopenms.dir/rule] Error 2
make: *** [pyopenms] Error 2
```

**Comment by mwalzer** [comment]
@hroest this was already failing before your fixes ... do you have any clue what might cause this?

**Comment by hroest** [comment]
@mwalzer which version of autowrap is this? This also builds on my machine with autowrap 0.6.1

**Comment by mwalzer** [comment]
autowrap was 0.5 and pyopenms built at some point on that machine...that is what puzzled me, alongside with that it built on another of my linux machines. So it builds with autowrap 0.7 - thanks.

Is there somewhere **documented** which versions of the nose and autowrap pyopenms depends on?
Also, if confident, pyopenms will always build with the newest versions, you might consider changing ```pip install ...``` in the documentation page to  ```pip install -U ...```
Just a wild idea: could you invoke the build of pyopenms via the setup.py? Because there, you can set version dependencies, right?

btw still get alot of warnings like ```warning: pyopenms/pyopenms.pyx:27101:186: Unreachable code```

**Comment by hroest** [comment]
>Is there somewhere documented which versions of the nose and autowrap pyopenms depends on?

yes, this would usually be checked during `cmake`, see https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/CMakeLists.txt#L125 but it seems that they are outdated at the moment (fix in #1730). The same is true for nose/numpy etc and all the other dependencies.

>  Just a wild idea: could you invoke the build of pyopenms via the setup.py? Because there, you can set version dependencies, right?

I think you can, this is basically what cmake does: https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/CMakeLists.txt#L410

> btw still get alot of warnings like warning: pyopenms/pyopenms.pyx:27101:186: Unreachable code

I get those too, that may be an issue with Cython and the code generated by it.
